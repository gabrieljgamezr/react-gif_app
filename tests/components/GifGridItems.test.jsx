import { render, screen } from "@testing-library/react";
import { GifGridItems } from "../../src/components/GifGridItems";

describe('Pruebas en <GifGridItems />', () => {

    const title = 'Título de prueba';

    const url = 'https://urldeprueba.com/';

    test('Debe hacer match con el snapshot', () => {

        const {container} = render(<GifGridItems title={title} url={url} />);
        
        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifGridItems title={title} url={url} />);
        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('Debe de mostrar el título en el componente', () => {

        render(<GifGridItems title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();

    });

});