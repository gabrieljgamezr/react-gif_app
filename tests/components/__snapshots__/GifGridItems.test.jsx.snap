// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItems /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Título de prueba"
      src="https://urldeprueba.com/"
    />
    <p>
      Título de prueba
    </p>
  </div>
</div>
`;
